### PR TITLE
dma - Fix compilation errors in DragonFly BSD

### PR DESCRIPTION
**Pull request by tuxillo**
Please check

### CODE REVIEW

File: dma.c [hunk 1]
```diff
@@ -60,6 +60,8 @@
 #include "dma.h"
 
 
+extern int yyparse(void);
```
Comment (corecode) [inline dma.c]: why not in the header?

File: aliases_scan.l [hunk 2]
```diff
@@ -5,7 +5,6 @@
 
 #define YY_NO_INPUT
 
-int yylex(void);
```
Comment (corecode) [inline aliases_scan.l]: what is wrong with this?

**Comment by tuxillo** [comment]
antonioh@andromeda:/s/dma$ make -f BSDmakefile
Warning: Object directory not changed from original /s/dma
yacc -d -o aliases_parse.c aliases_parse.y
cc  -O -pipe  -I/s/dma -DHAVE_REALLOCF -DHAVE_STRLCPY -DHAVE_GETPROGNAME -DLIBEXEC_PATH='"/usr/local/libexec"' -DDMA_VERSION='"v0.9.10.gb1056e4"' -DCONF_PATH='"/usr/local/etc/dma"'  -O0 -g -std=gnu99 -Wmissing-include-dirs -Wsystem-headers -Werror -Wall -Wformat-security -Winit-self -Wno-pointer-sign -Wextra -Wno-unused-parameter -Wstrict-prototypes -Wmissing-prototypes -Wpointer-arith -Wold-style-definition -Wreturn-type -Wcast-qual -Wwrite-strings -Wswitch -Wshadow -Wcast-align -Wunused-parameter -Wchar-subscripts -Winline -Wnested-externs -Wredundant-decls -c aliases_parse.c
aliases_parse.c:93:12: error: redundant redeclaration of 'yyparse' [-Werror=redundant-decls]
In file included from aliases_parse.y:6:0:
dma.h:176:5: note: previous declaration of 'yyparse' was here
cc1: all warnings being treated as errors
*** Error code 1

**Comment by corecode** [comment]
I find -Werror=redundant-decls to be too aggressive.  Is there a good reason to change this?  I would prefer having all declarations in the headers.

**Comment by tuxillo** [comment]
Well, not really. The only reason is to compile dma without having to specify NO_WERROR= in the command line.

**Comment by corecode** [comment]
So WARNS=6 means something different now?  Maybe we should downgrade the
warns then?

**Comment by tuxillo** [comment]
What do you mean?
-Wredundant-decls is in WARNS=6 for a while

**Comment by corecode** [comment]
Then we need to go back to WARNS=5, I guess.

**Comment by tuxillo** [comment]
What's the point having it on the headers or in the .c file as extern? I don't get it

**Comment by corecode** [comment]
Declarations should always be in a header, so that the declaration is checked when the definition is compiled.  That way you prevent the declaration diverging from the definition.

**Comment by tuxillo** [comment]
Then what's the point of the -Wredundant-decls in gcc? Even if it's too agressive, I don't see why there should be any redundant decls.

**Comment by corecode** [comment]
I don't know why DragonFly uses -Wredundant-decls.  I don't know what this useful for, except for cleaning code.

**Comment by corecode** [comment]
The redundant decls come from lex/yacc inserting the decls in the generated code.  Nothing we can do about it.

**Comment by tuxillo** [comment]
Okay then. If you don't want to move the decls then there's nothing to do here.

**Comment by corecode** [comment]
I guess reduce WARNS?

**Comment by tuxillo** [comment]
Nah, we'll keep moving the decls to the .c files by now. I find it more convenient :P 
Shall we close this pull request then?

**Comment by tuxillo** [comment]
But in any case, what's the point of the decl of yylex() in aliases_scan.l ?

**Comment by corecode** [comment]
it shouldn't be there, should go to the header

**Comment by tuxillo** [comment]
Ok cool, then I guess you'll move it. I'll close the pull request

**Pull request closed without merging**